projects/packages/sync/src/class-queue.php

```diff
@@ -46,14 +46,22 @@ public function __construct( $id ) {
 	public function add( $item ) {
 		global $wpdb;
 		$added = false;
+
+		// Attempt to serialize data, if an exception (closures) return early.
+		try {
+			$item = serialize( $item ); // phpcs:ignore WordPress.PHP.DiscouragedPHPFunctions.serialize_serialize
```

Comment:
Why not use `maybe_serialize` and avoid the PHPCS exception?

Reply:
I was thinking we'd hit the same error with maybe_serialize. I should make a unit test for this and test it out :)

```
function maybe_serialize( $data ) {
    if ( is_array( $data ) || is_object( $data ) ) {
        return serialize( $data );
    }
 
    /*
     * Double serialization is required for backward compatibility.
     * See https://core.trac.wordpress.org/ticket/12930
     * Also the world will end. See WP 3.6.1.
     */
    if ( is_serialized( $data, false ) ) {
        return serialize( $data );
    }
 
    return $data;
}
```

Reply:
Verified maybe_serialize runs into the same exception.